1-Draft/RFCNNNN-VSCode-PSScriptAnalyzer-Rules.md

```diff
@@ -0,0 +1,151 @@
+If a user wanted to opt-in for more rules,
+they have a few options:
+
+- Using the
+"PowerShell: Select PSScriptAnalyzer rules"
+command pallet action
+- Create a
+`PSScriptAnalyzerSettings.psd1`
+file with the rules you want and update your
+`powershell.scriptAnalysis.settingsPath`
+VSCode setting to point to that
+
+## Problem
+
+Using these mechanisms require understanding of what PSSA is,
+what rules are,
+how to configure PSSA,
+etc.
+
+It's a lot of information to go digging for.
+
+It should be easier to opt-in to more rules without needing to go through PSSA docs.
+
+## Solution - "Aggressive Analysis mode"
+
+I'm proposing an 
+"Aggressive Analysis mode"
+set of rules that you can flip on with one setting in VSCode. 
+In addition,
+adding a few other rules to the default rules.
+
+`powershell.scriptAnalysis.mode`
+
+This can be set to:
+
+- None - disables script analysis
+- Default - some crutial rules but nothing too crazy
+- Aggressive - more rules, more fun
+
+Here is the table of:
+
+- the current Defaults
+- the ones that will be a part of the "Aggressive analysis mode"
+- the ones we are considering to add to the Defaults (marked with an x)
+
+| Rule                                         | PSSA Type   | Mode       | Add to default |
+|----------------------------------------------|-------------|------------|----------------|
+| AlignAssignmentStatement                     | Warning     |            |                |
+| AvoidAssignmentToAutomaticVariable           | Warning     | Aggressive |                |
+| AvoidDefaultValueForMandatoryParameter       | Warning     | Default    |                |
+| AvoidDefaultValueSwitchParameter             | Warning     | Default    |                |
+| AvoidGlobalAliases*                          | Warning     | Aggressive | x              |
+| AvoidGlobalFunctions                         | Warning     | Aggressive | x              |
+| AvoidGlobalVars                              | Warning     | Aggressive | x              |
+| AvoidInvokingEmptyMembers                    | Warning     | Aggressive | x              |
+| AvoidNullOrEmptyHelpMessageAttribute         | Warning     |            |                |
+| AvoidShouldContinueWithoutForce              | Warning     | Aggressive |                |
+| AvoidUsingCmdletAliases                      | Warning     | Default    |                |
+| AvoidUsingComputerNameHardcoded              | Error       | Default    |                |
+| AvoidUsingConvertToSecureStringWithPlainText | Error       | Default    |                |
+| AvoidUsingDeprecatedManifestFields           | Warning     | Aggressive | x              |
+| AvoidUsingEmptyCatchBlock                    | Warning     |            |                |
+| AvoidUsingInvokeExpression                   | Warning     |            |                |
+| AvoidUsingPlainTextForPassword               | Warning     | Default    |                |
```

Comment:
I would not enable this by default but in the more thorough mode due to the same reasoning about `SecureString` not having enough support coverage by CI systems